subset.go

```diff
@@ -46,6 +48,17 @@ func checkSubset(expected, target reflect.Value, visited map[uintptr]*visit, dep
 		// fmt.Println("!target.IsValid()")
 		return false
 	}
+
+	switch expected.Kind() {
```

Comment:
Do we really need a switch statement if we only care about 1 case? Also, what happens if there is an error converting? What do we end up returning?

Reply:
have removed the switch statement, added that thinking we will be having multiple cases but i am sure now we won't. Secondly if we get an error, the condition below this switch will return false which check's if both have different types.